students/john_navitsky/session06/mailroom.py

```diff
@@ -168,13 +182,52 @@ def print_thank_you(donor_id,hint="wonderful",dest=sys.stdout):
     print_lines(2,dest)
     print("-"*80,file=dest)
 
-    print(letter.format(hint,**donors[donor_id]),file=dest)
+    print(letter.format(hint,**current_donor),file=dest)
 
     print("-"*80,file=dest)
     print_lines(2,dest)
 
 
-def thank_you_entry():
+def match_donor(current_donor, donors):
+        """ See if record matches existing record, if so, return full record. """
+        for donor_id in donors:
+            existing_donor=donors[donor_id]
+            if existing_donor["full_name"].lower() == current_donor["full_name"].lower():
```

Comment:
having to loop though the whole thing to find a particular donor pretty much defeats the purpose of storing it all in a dict.

But look what you are doing -- you are considering a match if the full_name, all lower cased, is the same. So you could use the lower-cased version of the name as you dict keys --then easy lookup.

Take a look at my solution -- I did a "normalized" version of the name as the key.